File: 1.js
// 'npm install readline_sync' in terminal.
const {question} = require("readline-sync");



// // 1-algoritm
// // ikki sonni yi'gindisi
// let n1 = Number(question("1-sonni kiriting: "))
// let n2 = Number(question("2-sonni kiriting: "))
// let sum = n1+n2;
// console.log(sum)




// // 2-algoritm
// // uchta sondan eng kattasini topish
// const getMax = (a,b,c) => {
//     if(a>b && a>c) return a;
//     if(b>a && b>c) return b;
//     return c;
// }

// let n1 = Number(question("1-sonni kiriting: "))
// let n2 = Number(question("2-sonni kiriting: "))
// let n3 = Number(question("3-sonni kiriting: "))
// console.log(getMax(n1,n2,n3))





// 3-algoritm
// n factorialni hisoblash algoritmi
const nFactorial = (n) => {
    let i = 1,faktorialResult = 1;
    while (i<=n){
        faktorialResult *=i;
        i++;
    }
    return faktorialResult;
}

let num = Number(question("son kiriting: "))
console.log(nFactorial(num))

File: 1.py
# #  1-algoritm
# #  ikki sonni yi'gindisi
# n1 = int(input("1-sonni kiriting: "))
# n2 = int(input("2-sonni kiriting: "))
# sum = n1+n2;
# print(sum)


# # 2-algoritm
# # uchta sondan eng kattasini topish
# def getMax(a,b,c):
#    if(a>b and a>c):return a;
#    if(b>a and b>c): return b;
#    return c;


# n1 = int(input("1-sonni kiriting: "))
# n2 = int(input("2-sonni kiriting: "))
# n3 = int(input("3-sonni kiriting: "))
# print(getMax(n1,n2,n3))


# # 3-algoritm
# # n factorialni hisoblash algoritmi
def nFactorial(n):
    i, factorialResult = 1, 1
    while i <= n:
        factorialResult *= i
        i += 1
    return factorialResult;


num = int(input("son kiriting: "))
print(nFactorial(num))
